### PR TITLE
ESP32 cellular integration

### PR DESCRIPTION
**Pull request by ferre111**

### CODE REVIEW

**Comment by ferre111** [comment]
@Mierunski @tkucma
In my opinion, a better solution would be to create two articles, the first one describe the use of previously prepared binary files, the second to describe the use of `idf.py`. In both of them describe the use of any ESP32 board. However, it will be inconsistent with other articles that were written for a specific board.

**Comment by mkrawczyk-avs** [comment]
I modified Step #1 and added a pic in Step #2.
Please check if that is ok, @ferre111

File: docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+## Step 1: Prepare project
+0. Create a project directory for the integration.
+0. Open a command line interface and run `git clone https://github.com/AVSystem/Anjay-esp32-client --recursive`.
+0. Run `. $HOME/esp/esp-idf/export.sh` and `idf.py set-target esp32`.
+0. Open the **menuconfig** with the `idf.py menuconfig` command, navigate to **Component config -> anjay-esp32-client**.
+    - In **Choose targeted development board**, select ``M5StickC`` from the list of supported boards.
+    - In **Choose an interface**, select `External BG96 module`.
+    - In **BG96 module configuration**, specify the UART port, Tx pin and Rx pin for the BG96 module. Example port and pin numbers are provided in the screenshot below.
+      ![BG96 UART example configuration](images/BG96_uart_config.png "BG96 UART example configuration")
+    - In **Client options**, provide device credentials:
```
Comment (ferre111) [inline docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md:20]:
```suggestion
    - In **Client options**, provide device credentials and Server URI:
```

File: docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+To connect your ESP32-DevKitC to the Coiote IoT Device Management LwM2M Server, use your access to a Coiote DM installation, or register at https://eu.iot.avsystem.cloud/ to get access.
+
+To connect the board:
+
+0. Log in to Coiote DM and from the left side menu, select **Device Inventory**.
+0. In **Device Inventory**, click **Add device**.
+0. Select the **Connect your LwM2M device directly via the Management server** tile.
+![Add via Mgmt](images/mgmt_tile.png "Add via Mgmt")
+0. In the **Device credentials** step:
+    - In the **Device ID** field, type the endpoint name provided in the `nvs_config.csv`, e.g. `ESP32_test`.
```
Comment (ferre111) [inline docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md:41]: All `nvs_config.csv` probably should be replace with `IDF menuconfig **Client options**` after your changes.

File: docs/Anjay_integrations/ESP32_integration/M5StickBG96.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+## Step 1: Prepare project
+0. Create a project directory for the integration.
+0. Open a command line interface and run `git clone https://github.com/AVSystem/Anjay-esp32-client --recursive`.
+0. Run `. $HOME/esp/esp-idf/export.sh` and `idf.py set-target esp32`.
+0. Open the **menuconfig** with the `idf.py menuconfig` command, navigate to **Component config -> anjay-esp32-client**.
+    - In **Choose targeted development board**, select ``M5StickC`` from the list of supported boards.
+    - In **Choose an interface**, select `External BG96 module`.
+    - In **BG96 module configuration**, specify the UART port, Tx pin and Rx pin for the BG96 module. Example port and pin numbers are provided in the screenshot below.
+      ![BG96 UART example configuration](images/BG96_uart_config.png "BG96 UART example configuration")
+    - In **Client options**, provide device credentials:
```
Comment (ferre111) [inline docs/Anjay_integrations/ESP32_integration/M5StickBG96.md:20]:
```suggestion
    - In **Client options**, provide device credentials and Server URI:
```

File: docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md [hunk 4]
```diff
@@ -6,18 +6,24 @@ Integrate your ESP32-based device with Quectel BG96 module to manage it via Coio
...
 0. Open a command line interface and run `git clone https://github.com/AVSystem/Anjay-esp32-client --recursive`.
-0. Open the `nvs_config.csv` file. Provide your credentials in [endpoint_name], [identity], [psk], [lwm2m_server_uri] (without the `[]` brackets). The other fields can remain unchanged.
 0. Run `. $HOME/esp/esp-idf/export.sh` and `idf.py set-target esp32`.
-0. Open menuconfig with the `idf.py menuconfig` command, navigate to the `Component config/anjay-esp32-client`, and from the supported boards, select **ESP32-DevKitC and derivatives**.
-0. Go to the `Choose an interface`, and select `External BG96 module`.
-0. Configure UART port for the BG96 module in the `BG96 module configuration`.
-![BG96 UART example configuration](images/BG96_uart_config.png.png "BG96 UART example configuration")
-0. Configure PDN in the `Connection configuration`.
+0. Open the **menuconfig** with the `idf.py menuconfig` command, navigate to **Component config -> anjay-esp32-client**.
+    - In **Choose targeted development board**, select ``M5StickC`` from the list of supported boards.
```
[the remark this answers is not in the document]
Comment (ferre111) [reply docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md:16]: Also this double apostrophe is intentional?

File: docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    - In the **Device ID** field, type the endpoint name provided in the `nvs_config.csv`, e.g. `ESP32_test`.
+    - In the **Security mode** section, select the **PSK** mode.
+    - In the **Key identity** field, type the identity provided in the `nvs_config.csv`, e.g. `ESP32_test`.
+    - In the **Key** field, type the `psk` key provided in the `nvs_config.csv`.
+    ![Device credentials step](images/add_mgmt_quick.png "Device credentials step")
+0. Click the **Add device** button and **Confirm** in the confirmation pop-up.
+0. In the **Connect your device** step, the server is waiting for the board to connect. You can now start connecting the device.
+
+## Step 4: Flash ESP32-DevKitC
+0. Open a command line interface and run `idf.py -b 750000 flash monitor`.
```
Comment (ferre111) [inline docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md:50]:
```suggestion
0. Open the command line interface, go to your project directory, and run `idf.py flash monitor`.
```

**Review by ferre111 — COMMENTED** [review]
Reviewed. @mkrawczyk-avs

File: docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (mkrawczyk-avs) [reply docs/Anjay_integrations/ESP32_integration/ESP32-DevKitCBG96.md:16]: it's not, but it doesn't do anthing special

**Pull request merged**